core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -326,24 +326,25 @@ private ConnectivityState getAggregatedState() {
       Subchannel maybeRegister(
           String stickinessValue, @Nonnull Subchannel subchannel, List<Subchannel> rrList) {
         final Ref<Subchannel> newSubchannelRef = subchannel.getAttributes().get(STICKY_REF);
-        Ref<Subchannel> existingSubchannelRef = null;
+        Ref<Subchannel> existingSubchannelRef;
         while (true) {
-          if (existingSubchannelRef != null) {
-            if (stickinessMap.replace(stickinessValue, existingSubchannelRef, newSubchannelRef)) {
-              return subchannel;
-            }
-            existingSubchannelRef = stickinessMap.get(stickinessValue);
-          } else if ((existingSubchannelRef =
-                stickinessMap.putIfAbsent(stickinessValue, newSubchannelRef)) == null) {
+          if ((existingSubchannelRef =
+              stickinessMap.putIfAbsent(stickinessValue, newSubchannelRef)) == null) {
```

Comment:
nit: this might have better readability

```java
existingSubchannelRef = stickinessMap.putIfAbsent(stickinessValue, newSubchannelRef);
if (existingSubchannelRef == null) {
```

Reply:
Changed!